### PR TITLE
vndr: migrate to buildGoModule

### PR DESCRIPTION
**Pull request by msanft**
## Description of changes

Migrating to `buildGoModule` so `buildGoPackage` can be deprecated as per https://github.com/NixOS/nixpkgs/issues/318069. Project needs an upstream patch to do so, as it does not have a vendor directory, nor is a module.

Also moved to `by-name`.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by msanft** [comment]
@vdemeester I think this might should've gone to https://github.com/NixOS/nixpkgs/pull/318922

**Comment by vdemeester** [comment]
> @vdemeester I think this should've gone to #318922

Indeed.. wrong tab.. 🤦🏼

File: pkgs/by-name/vn/vndr/package.nix [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  vendorHash = null;
+
+  patches = [
+    # Migrate the project to Go modules.
+    (fetchpatch {
+      url = "https://github.com/LK4D4/vndr/commit/9f7c8344a3114d96ad2ef43d070a731285a23dc0.patch";
+      sha256 = "sha256-AzsnuOWeXVOelvdBcOzqn1A5L7qcg77mRvB2C3UpxqI=";
+    })
+  ];
+
```
Comment (katexochen) [inline pkgs/by-name/vn/vndr/package.nix:27]:
```suggestion
```

File: pkgs/by-name/vn/vndr/package.nix [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  pname = "vndr";
+  version = "0.1.2-unstable-2022-12-29";
+
+  src = fetchFromGitHub {
+    owner = "LK4D4";
+    repo = "vndr";
+    rev = "87603e47e8ea2ddac96f508fc9e9d6fc17b198b0";
+    sha256 = "sha256-L7OemAPCv7epOVmjrDDtiGqQqzscm5zj3C6dsZP4uUc=";
+  };
+
```
Comment (katexochen) [inline pkgs/by-name/vn/vndr/package.nix:17]:
```suggestion

  postPatch = ''
    go mod init github.com/LK4D4/vndr
  '';
```
Patch isn't really needed, we can just init the module postPatch.

**Review by katexochen — CHANGES_REQUESTED** [review]

**Review by katexochen — APPROVED** [review]

**Pull request merged**